src/control/states/HoldingState.ts

```diff
@@ -36,17 +36,15 @@ class HoldingState extends State {
 
   public onChange(ctx: Parameters<State["onChange"]>[0]): void {
     const { flicking, axesEvent, transitTo } = ctx;
-
+    const animatingContext = flicking.control.controller.animatingContext;
     const inputEvent = axesEvent.inputEvent as { offsetX: number; offsetY: number };
+    const offset = flicking.horizontal ? inputEvent?.offsetX : inputEvent?.offsetY;
 
-    const offset = flicking.horizontal
-      ? inputEvent.offsetX
-      : inputEvent.offsetY;
 
     const moveStartEvent = new ComponentEvent(EVENTS.MOVE_START, {
       isTrusted: axesEvent.isTrusted,
       holding: this.holding,
-      direction: getDirection(0, -offset),
+      direction: inputEvent ? getDirection(0, -offset) : getDirection(animatingContext.start, animatingContext.end),
```

Comment:
The fix may be okay, but I'm a bit concerned about triggered events.
In the past PR, you've fixed the way that triggers the `holdEnd` event, while this doesn't.
So, for the two same use cases, events are triggered in a different way.
1. `holdStart` -> `holdEnd` -> `moveStart` (previous PR)
2. `holdStart` -> `moveStart` (current PR)

Please check the actual cause why the previous PR doesn't resolve the issue, and fix this PR in more proper way.